File: src/main/java/leets/leenk/domain/media/domain/repository/MediaRepository.java
package leets.leenk.domain.media.domain.repository;

import java.util.List;
import java.util.Optional;
import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.leenk.domain.entity.Leenk;
import leets.leenk.domain.media.domain.entity.Media;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MediaRepository extends JpaRepository<Media, Long> {

    List<Media> findAllByFeedInOrderByPosition(List<Feed> feeds);

    List<Media> findAllByFeedOrderByPosition(Feed feed);

    List<Media> findAllByLeenkOrderByPosition(Leenk leenk);

    Optional<Media> findFirstByLeenkOrderByPositionAsc(Leenk leenk);

    List<Media> findAllByLeenkInOrderByPosition(List<Leenk> leenks);
}

File: src/main/kotlin/leets/leenk/domain/media/domain/repository/MediaRepository.kt
package leets.leenk.domain.media.domain.repository

import leets.leenk.domain.feed.domain.entity.Feed
import leets.leenk.domain.leenk.domain.entity.Leenk
import leets.leenk.domain.media.domain.entity.Media
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface MediaRepository : JpaRepository<Media, Long> {
    fun findAllByFeedInOrderByPosition(feeds: List<Feed>): List<Media>

    fun findAllByFeedOrderByPosition(feed: Feed): List<Media>

    fun findAllByLeenkOrderByPosition(leenk: Leenk): List<Media>

    fun findFirstByLeenkOrderByPositionAsc(leenk: Leenk): Optional<Media>

    fun findAllByLeenkInOrderByPosition(leenks: List<Leenk>): List<Media>

    fun deleteAllByFeed(feed: Feed)

    fun findByMediaUrl(originalUrl: String): Optional<Media>
}
